agentrs/src/session.rs

```diff
@@ -0,0 +1,58 @@
+use uuid::Uuid;
+
+// Binary package format:
+// [ sid(16 bytes) | len(4 bytes) | payload(len bytes) ]
+// a 20 bytes header with a UUID and a length field,
+// followed by the payload of the specified length.
+#[derive(Debug, Clone, Copy)]
+pub struct Header {
```

Comment:
Add a `dataSize` field instead of returning in a tuple; it's still header information.